### ISSUE TITLE
Expose way for user to re-link account to Github

### ISSUE THREAD

**Issue by amisevsk** [openshiftio/openshift.io#390]
If a user revokes the OSIO permission on Github, it's currently not clear how to re-link their account. There should be a button (possibly on the "update profile" page) with an option to relink. When the permission is revoked, codebases are broken and workspaces cannot be started.

Currently afaict it's only possible by going through sso.openshift.io, which is not clear at all.

**Comment by joshuawilson** [comment]
This is a really good point. For now the answer is to go to sso.openshift.io/auth/realms/fabric8/account/identity

But we should build something into the system to support this.

**Comment by aslakknutsen** [comment]
The system should really auto detect it and inform the user on need. Similar to this; https://github.com/fabric8-ui/fabric8-ui/issues/1405 tho not 100% sure why this one is closed. (see discussion at the end)

**Comment by joshuawilson** [comment]
I had thought it would block access to the site if you didn't have it. 
But if we want the system to be accessable, then we should disable the pages that need github access until they have it granted. That would mean going back in and setting it. 
cc @catrobson

**Comment by aslakknutsen** [comment]
@joshuawilson The systems that require additional linking etc should really request that on use, not up front like we now. e.g. we don't need oso or github token to do planning.

**Comment by joshuawilson** [comment]
I remember talking about that. I agree and would prefer to request is when needed. That gives the user the reason for the access and it makes the getting started experience cleaner and faster.
What do you think @qodfathr ?

**Comment by qodfathr** [comment]
I agree that only asking for access when first needed makes sense. But this is not a priority to fix relative to other open concerns, so treat accordingly.

**Comment by joshuawilson** [comment]
I'm glad we have agreed to the correct work flow. However, we still have a bug that needs to be fixed. If a new user doesn't agree to github access and they are still allowed to login, the system will fail for them and they won't know why.

**Comment by amisevsk** [comment]
@joshuawilson For what it's worth, while my Github token was revoked, I would get "Could not connect to Github Account" as an error message in the corner, so it didn't take me long to figure out what the issue was. The codebases view was however mostly blank, so I suppose explicitly requesting access instead of erroring is better.

**Comment by catrobson** [comment]
@dlabrecq I thought for a while we were redirecting to the getting started page when we recognized that the github token was needed for the user to reconnect.  Is that something we could put back into place for now until we can get the full expected user flows defined and implemented?

**Comment by dlabrecq** [comment]
When the user logs in, and one or both tokens are not linked, they land on the getting started page. We had duplicate controls to link tokens via the update profile page, but we removed it.

The problem with that is; when the UI detects a token changed, it logs the user out. Therefore, users never gets to use these controls on the profile page. This is why they were removed.

Unless you're already on the update profile page and revoke access via Keycloak, for example. And that must be done before the UI detects a change and logs you out.
